src/main/java/io/jenkins/plugins/postgresql/PostgreSQLFingerprintStorageDescriptor.java

```diff
@@ -0,0 +1,143 @@
+ * THE SOFTWARE.
+ */
+package io.jenkins.plugins.postgresql;
+
+import com.cloudbees.plugins.credentials.CredentialsMatchers;
+import com.cloudbees.plugins.credentials.CredentialsProvider;
+import com.cloudbees.plugins.credentials.common.StandardListBoxModel;
+import com.cloudbees.plugins.credentials.common.StandardUsernamePasswordCredentials;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.model.Item;
+import hudson.security.ACL;
+import hudson.util.FormValidation;
+import hudson.util.ListBoxModel;
+import jenkins.fingerprints.FingerprintStorageDescriptor;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+import org.kohsuke.stapler.AncestorInPath;
+import org.kohsuke.stapler.QueryParameter;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+
+import javax.servlet.ServletException;
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.Collections;
+
+public class PostgreSQLFingerprintStorageDescriptor extends FingerprintStorageDescriptor {
+
+    public static final String DEFAULT_HOST = "localhost";
+    public static final int DEFAULT_PORT = 5432;
+    public static final String DEFAULT_DATABASE_NAME = "defaultDB";
+    public static final boolean DEFAULT_SSL = false;
+    public static final int DEFAULT_CONNECTION_TIMEOUT = 2000;
+    public static final int DEFAULT_SOCKET_TIMEOUT = 2000;
+    public static final String DEFAULT_CREDENTIALS_ID = "";
+
+    @Override
+    public @NonNull String getDisplayName() {
+        return Messages.PostgreSQLFingerprintStorage_DisplayName();
+    }
+
+    @RequirePOST
+    @Restricted(NoExternalUse.class)
+    public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item item, @QueryParameter String credentialsId) {
+        StandardListBoxModel result = new StandardListBoxModel();
+        if (item == null) {
+            if (!Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
+                return result.includeCurrentValue(credentialsId);
+            }
+        } else {
+            if (!item.hasPermission(Item.EXTENDED_READ) && !item.hasPermission(CredentialsProvider.USE_ITEM)) {
+                return result.includeCurrentValue(credentialsId);
+            }
+        }
+        return result
+                .includeEmptyValue()
+                .includeMatchingAs(
+                        ACL.SYSTEM,
+                        Jenkins.get(),
+                        StandardUsernamePasswordCredentials.class,
+                        Collections.emptyList(),
+                        CredentialsMatchers.always()
+                )
+                .includeCurrentValue(credentialsId);
+    }
+
+    @Restricted(NoExternalUse.class)
+    public FormValidation doCheckCredentialsId(@AncestorInPath Item item, @QueryParameter String value) {
+        if (item == null) {
```

Comment:
You can simplify this and take some nesting out by doing:
```if (item == null && !Jenkins.get().hasPermission(Jenkins.ADMINISTER)) {
  return FormValidation.ok();
}
```
You can then remove the else block.